gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/MySqlDagManagementStateStore.java

```diff
@@ -116,128 +112,78 @@ public void removeFlowSpec(URI uri, Properties headers, boolean triggerListener)
     this.flowCatalog.remove(uri, headers, triggerListener);
   }
 
-  public synchronized void setTopologySpecMap(Map<URI, TopologySpec> topologySpecMap) throws IOException {
+  public synchronized void setTopologySpecMap(Map<URI, TopologySpec> topologySpecMap) {
     this.topologySpecMap = topologySpecMap;
     start();
   }
 
-  private DagStateStore createDagStateStore(Config config, Map<URI, TopologySpec> topologySpecMap) {
+  private DagStateStoreWithDagNodes createDagStateStore(Config config, Map<URI, TopologySpec> topologySpecMap) {
     try {
-      Class<?> dagStateStoreClass = Class.forName(ConfigUtils.getString(config, DAG_STATESTORE_CLASS_KEY, MysqlDagStateStore.class.getName()));
-      return (DagStateStore) GobblinConstructorUtils.invokeLongestConstructor(dagStateStoreClass, config, topologySpecMap);
+      Class<?> dagStateStoreClass = Class.forName(ConfigUtils.getString(config, DAG_STATESTORE_CLASS_KEY, MysqlDagStateStoreWithDagNodes.class.getName()));
+      return (DagStateStoreWithDagNodes) GobblinConstructorUtils.invokeLongestConstructor(dagStateStoreClass, config, topologySpecMap);
     } catch (ReflectiveOperationException e) {
       throw new RuntimeException(e);
     }
   }
 
   @Override
-  public void checkpointDag(Dag<JobExecutionPlan> dag) throws IOException {
```

Comment:
this api will no longer be used for "checkpointing", will be used just for adding the dag for once in the beginning for it's lifecycle (by launch and resume dag procs)

Reply:
I think this makes sense primarily if the dag is immutable (i.e. you are certain when calling this line that you are creating a new row and adding a dag that already exists may result in an error) otherwise checkpoint may still make sense here semantically.

Reply:
actually with the new table, only dag nodes will be added in the table. (so far we add complete dag as one json)
all the mutable operations happen only at dag node level.